src/cart/crt_iv.c

```diff
@@ -1920,15 +1920,12 @@ handle_ivsync_response(const struct crt_cb_info *cb_info)
 		iv_ops->ivo_on_put(iv_sync->isc_ivns_internal, NULL,
 					iv_sync->isc_user_priv);
 		D_FREE(iv_sync->isc_iv_key.iov_buf);
+	}
 
```

Comment:
isc_do_callback can only be set if iv_sync->isc_ivns_internal is set and vice versa. why put it on a separate condition?

Reply:
It seems so indeed. I'm just trying to simplify the reasoning (e.g., one doesn't have to figure out that isc_do_callback == delay_completion && "the isc_ivns_internal reference is taken only if delay_completion == true"). I tend to feel the leaks might be due to the complexity underlying the current reference management approach.

Reply:
I feel in opposite that it can complicate reading by adding another condition. instead in my view the code that performs addref should be just changed to if (iv_sync_cb->isc_do_callback) instead of if (delay_completion). This way conditions for taking and releasing reference count are symmetrical. 

I do agree with simplifications done in other parts of code though, so its only this one that i have issues with

Reply:
I'll change this place back and assert "if isc_do_callback == false then isc_ivns_internal must be NULL" instead.

Reply:
thanks